iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -632,7 +632,12 @@ private async Task OpenAsyncInternal(CancellationToken cancellationToken)
             if (IsProxyConfigured())
             {
                 //No need to do a DNS lookup since we have the proxy address already
+
+#if NET451
                 _serverAddresses = new IPAddress[0];
+#else
+                _serverAddresses = Array.Empty<IPAddress>();
```

Comment:
CA182 - Avoid unnecessary zero-length array allocations.  Use `Array.Empty<IPAddress>()` instead. (not available in net451)